stitchclient/client.py

```diff
@@ -155,9 +155,12 @@ def _send_batch(self, batch):
         for body, callback_args in partition_batch(batch, self.max_batch_size_bytes):
             self._send(body, callback_args)
 
-        self.target_messages_per_batch = min(self.max_messages_per_batch,
-                                             0.8 * (self.max_batch_size_bytes / self.moving_average_bytes_per_record()))
-
+        if self.moving_average_bytes_per_record():
+            self.target_messages_per_batch = \
+                min(self.max_messages_per_batch,
+                    0.8 * (self.max_batch_size_bytes / self.moving_average_bytes_per_record()))
+        else:
+            self.target_messages_per_batch = self.max_messages_per_batch
```

Comment:
Expanding on my comment above, maybe here instead of using the return value of `moving_average_bytes_per_record` to predicate how we set `target_messages_per_batch`, we should instead explicitly check that `num_records` is nonzero. I.e.

```python
        if self.num_records > 0:
            self.target_messages_per_batch = \
                min(self.max_messages_per_batch,
                    0.8 * (self.max_batch_size_bytes / self.moving_average_bytes_per_record()))
        else:
            self.target_messages_per_batch = self.max_messages_per_batch
```

Also just curious -- why do we set `target_messages_per_batch` to the max when `num_records` is 0?

Reply:
so I wasn't sure what to do if num_records = 0 here.  I was faced with what to do with min(n, Exception) and so just picked n